src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -216,23 +216,23 @@ private function storeValue(string $propertyName, array &$mapToProperties, array
     /**
      * @param callable(): mixed $fn
      */
-    private function call(callable $fn, mixed $value, object $object): mixed
+    private function call(callable $fn, mixed $value, object $source, ?object $target = null): mixed
```

Comment:
To make sure we don't forget to inject `$target` when necessary, I would make this parameter required, with `null` explicitly set where this method is called.